include/powerloader/curl.hpp

```diff
@@ -35,17 +36,25 @@ namespace powerloader
         bool m_serious;
     };
 
-    struct POWERLOADER_API Response
+    struct POWERLOADER_API BaseResponse
     {
-        std::map<std::string, std::string> header;
-        mutable std::stringstream content;
+        std::map<std::string, std::string> headers;
 
-        curl_off_t avg_speed;
+        curl_off_t average_speed;
         curl_off_t downloaded_size;
         long http_status;
         std::string effective_url;
 
         bool ok() const;
+
+        tl::expected<std::string, std::out_of_range> get_header(const std::string& header) const;
+
+        void fill_values(CURLHandle& handle);
+    };
+
+    struct Response : public BaseResponse
```

Comment:
Why splitting the Reponse class? If that's required, then the semantics should be clarified (at least protected constructors and destructors in the base class so that it clears that it is not intended to be used as a polymorphic class).